src/View/ViewableData.php

```diff
@@ -562,9 +567,18 @@ public function obj($fieldName, $arguments = [], $cache = false, $cacheName = nu
         if (!is_object($value)) {
             // Force cast
             $castingHelper = $this->castingHelper($fieldName);
-            $valueObject = Injector::inst()->create($castingHelper, $fieldName);
-            $valueObject->setValue($value, $this);
-            $value = $valueObject;
+            // Cast arrays to ArrayList by default
+            if (is_array($value) && $castingHelper === $this->config()->get('default_cast')) {
+                $value = ArrayList::create($value);
+            } else {
+                $valueObject = Injector::inst()->create($castingHelper, $fieldName);
+                $valueObject->setValue($value, $this);
+                $value = $valueObject;
+            }
+        }
+
+        if (is_a($value, ArrayList::class)) {
+            $this->convertArrayListItemsToViewableData($value);
         }
```

Comment:
What about other list types? We have to assume someone has made other implementations of `SS_List`.